### PR TITLE
Better tasking less locks

### PR DESCRIPTION
**Pull request by joaoantoniocardoso**
merge after #303

### CODE REVIEW

File: src/server/manager.rs [hunk 1]
```diff
@@ -67,25 +66,26 @@ pub async fn run(server_address: &str) -> Result<(), std::io::Error> {
...
+            //                 InMemoryBackend::builder().build(),
+            //                 SimpleInputFunctionBuilder::new(std::time::Duration::from_secs(1), 4)
+            //                     .real_ip_key()
+            //                     .build(),
+            //             )
+            //             .add_headers()
+            //             .build(),
+            //         )
+            //         .route("", web::get().to(pages::thumbnail)),
+            // )
```
Comment (joaoantoniocardoso) [inline src/server/manager.rs:83]: This needs to be worked with care to work with only one worker...

File: src/server/pages.rs [hunk 2]
```diff
@@ -380,41 +382,41 @@ pub fn sdp(sdp_file_request: web::Query<SdpFileRequest>) -> HttpResponse {
...
+//         Some(Err(error)) => HttpResponse::ServiceUnavailable()
+//             .reason("Thumbnail temporarily unavailable")
+//             .insert_header((header::RETRY_AFTER, 10))
+//             .content_type("text/plain")
+//             .body(format!(
+//             "Thumbnail for source {:?} is temporarily unavailable. Try again later. Details: {error:?}",
+//             thumbnail_file_request.source
+//         )),
+//     }
+// }
```
Comment (joaoantoniocardoso) [inline src/server/pages.rs:422]: This needs to be reworked with care to work with only one worker...

File: src/stream/mod.rs [hunk 3]
```diff
@@ -154,27 +156,33 @@ impl Stream {
...
+                let new_state = match StreamState::try_new(
+                    &video_and_stream_information,
+                    &pipeline_id,
+                )
+                .await
+                {
+                    Ok(state) => state,
+                    Err(error) => {
+                        error!("Failed to recreate the stream {pipeline_id:?}: {error:#?}. Trying again in one second...");
+                        std::thread::sleep(std::time::Duration::from_secs(1));
```
Comment (joaoantoniocardoso) [inline src/stream/mod.rs:174]:
```suggestion
                        tokio::time::sleep(tokio::time::Duration::from_secs(1));
```

**Comment by patrickelectric** [comment]
> > The threads do not finish..
> > image
> > A shutdown mechanism like #314 is still necessary
> 
> Even if you add the shutdown mechanism on top of this patch, your test will fail. If you have a while loop inside the drop, you need to allow more workers.

Indeed the method does not work with this approach, I'll check if there other ways for us to investigate this.

**Comment by joaoantoniocardoso** [comment]
Superposed by #377

**Pull request closed without merging**